src/tokens/saga.ts

```diff
@@ -217,6 +223,50 @@ export function* fetchTokenBalancesSaga() {
   }
 }
 
+export async function fetchTokenPriceHistory(
+  tokenId: string,
+  startTimestamp?: number,
+  endTimestamp?: number
+): Promise<TokenPriceHistoryEntry[]> {
+  const queryParams = new URLSearchParams()
+  if (startTimestamp) {
+    queryParams.append('startTimestamp', `${startTimestamp}`)
+  }
+  if (endTimestamp) {
+    queryParams.append('endTimestamp', `${endTimestamp}`)
+  }
+
+  const url = `${
+    networkConfig.blockchainApiUrl
+  }/tokensInfo/${tokenId}/priceHistory?${queryParams.toString()}`
+  const response = await fetchWithTimeout(url)
+  if (!response.ok) {
+    throw new Error(
+      `Failed to fetch price history for ${tokenId}: ${response.status} ${response.statusText}`
+    )
+  }
+  return await response.json()
+}
+
+export function* fetchTokenPriceHistorySaga({
+  payload: { tokenId, startTimestamp, endTimestamp },
+}: ReturnType<typeof fetchPriceHistoryStart>) {
+  try {
+    if (!getFeatureGate(StatsigFeatureGates.USE_PRICE_HISTORY_FROM_BLOCKCHAIN_API)) {
```

Comment:
should we also just avoid dispatching that action if this feature gate is false?

Reply:
where is this action dispatched anyway?

Reply:
It is not dispatched yet. I planned to do it in the reworked component on mount.